src/distilabel/steps/tasks/magpie/base.py

```diff
@@ -0,0 +1,267 @@
+# Copyright 2023-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Union
+
+from pydantic import Field, PositiveInt
+
+from distilabel.llms.base import LLM
+from distilabel.llms.mixins.magpie import MagpieChatTemplateMixin
+from distilabel.mixins.runtime_parameters import (
+    RuntimeParameter,
+    RuntimeParametersMixin,
+)
+from distilabel.steps.base import StepInput
+from distilabel.steps.tasks.base import Task
+
+if TYPE_CHECKING:
+    from distilabel.steps.tasks.typing import ChatType, FormattedInput
+    from distilabel.steps.typing import StepOutput
+
+MAGPIE_MULTI_TURN_SYSTEM_PROMPT = (
+    "You are a helpful Al assistant. The user will engage in a multi−round conversation"
```

Comment:
Maybe this should be changed?
```suggestion
    "You are a helpful Al assistant. The user will engage in a multi−turn conversation"
```

Reply:
I copy pasted it from the paper 😋